pyfolio/utils.py

```diff
@@ -259,3 +260,45 @@ def extract_rets_pos_txn_from_zipline(backtest):
     transactions.index.tz = None
 
     return returns, positions, transactions, gross_lev
+
+
+SETTINGS = {
+    # Settings dict to store functions/values that may
+    # need to be overridden depending on the users environment
+    'returns_func': default_returns_func
+}
+
+
+def register_return_func(func):
+    """
+    Registers the function that will be called for
+    retrieving returns data.
+
+    Parameters
+    ----------
+    func : function
+        A function that accepts a symbol and
+        returns a pandas.Series of asset returns.
+
+    Returns
+    -------
+    None
+    """
+    SETTINGS['returns_func'] = func
+
+
+def get_symbol_rets(symbol):
```

Comment:
Yeah, that's my thinking. While theoretically it'd be nice to set an API explicitly I think this will clash with reality where people might have to pass in certain arguments (credentials are a good example).